src/components/token/helper/create-id-token.ts

```diff
@@ -1,54 +1,83 @@
-import { createHash } from "crypto";
+import { createHash, randomBytes } from "crypto";
 import { Config } from "../../../config";
 import {
   ID_TOKEN_EXPIRY,
+  INVALID_ISSUER,
   ISSUER_VALUE,
+  oneDayInSeconds,
   SESSION_ID,
   TRUSTMARK_URL,
 } from "../../../constants";
 import { logger } from "../../../logger";
 import AuthRequestParameters from "src/types/auth-request-parameters";
 import { signToken } from "./sign-token";
 import { IdTokenClaims } from "../../../types/id-token-claims";
+import { fakeSignature } from "../../../components/utils/fake-signature";
+import { makeHeaderInvalid } from "../../utils/make-header-invalid";
 
 export const createIdToken = async (
   authRequestParams: AuthRequestParameters,
   accessToken: string
 ): Promise<string> => {
   logger.info("Creating Id token");
+  const config = Config.getInstance();
 
-  const clientConfig = Config.getInstance();
   const idTokenClaims = createIdTokenClaimSet(
-    clientConfig,
+    config,
     authRequestParams,
     accessToken
   );
 
-  const signedIdToken = await signToken(idTokenClaims);
+  const idTokenErrors = config.getIdTokenErrors();
+
+  let signedIdToken = await signToken(idTokenClaims);
+
+  if (idTokenErrors.includes("INVALID_ALG_HEADER")) {
+    signedIdToken = makeHeaderInvalid(signedIdToken);
+  }
+  if (idTokenErrors.includes("INVALID_SIGNATURE")) {
+    signedIdToken = fakeSignature(signedIdToken);
+  }
 
   logger.info("Id token created");
 
   return signedIdToken;
 };
 
 const createIdTokenClaimSet = (
-  clientConfig: Config,
+  config: Config,
   authRequestParams: AuthRequestParameters,
   accessToken: string
 ): IdTokenClaims => {
-  const iat = Math.floor(Date.now() / 1000);
-  const exp = iat + ID_TOKEN_EXPIRY;
+  const idTokenErrors = config.getIdTokenErrors();
+  const timeNow = Math.floor(Date.now() / 1000);
+  const iat = idTokenErrors.includes("TOKEN_NOT_VALID_YET")
+    ? timeNow + oneDayInSeconds
+    : timeNow;
+  const exp = idTokenErrors.includes("TOKEN_EXPIRED")
+    ? timeNow - oneDayInSeconds
+    : iat + ID_TOKEN_EXPIRY;
```

Comment:
could there be some conflict here if we have both errors? I think we want this to be `timeNow + ID_TOKEN_EXPIRY`?

Reply:
Good spot, have fixed!